pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -557,6 +557,30 @@ func (r *ReconcileClusterDeployment) reconcile(request reconcile.Request, cd *hi
 		return reconcile.Result{}, nil
 	}
 
+	// Ensure that if DNSZone already exists, it's PreserveOnDelete setting matches the ClusterDeployment:
```

Comment:
Is it possible to use `ensureManagedDNSZone` here, with maybe a bool stating if managed dns zone not found then don't create it?

Reply:
I don't think it's necessary to run the same code twice, but I am wondering if we should just move ensureManagedDNSZone call up here, it would help repair a missing dnszone for an existing cluster, which wouldn't happen today. Checking if this is feasible.